src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -71,6 +71,38 @@ public void removeUser(TaskData task, long userID) {
     }
   }
 
+  public void setComplete(long taskID) {
+    setComplete(getTaskByID(taskID));
+  }
+
+  public void setComplete(TaskData task) {
+    ArrayList<TaskData> subtasks = getSubtasks(task);
+    for (TaskData subtask : subtasks) {
+      if (subtask.getStatus() != Status.COMPLETE) {
+        return;
+      }
+    }
+    task.setStatus(Status.COMPLETE);
+    if (task.getTaskID() != 0) {
+      datastore.put(task.toEntity());
+    }
+  }
+
+  public void setIncomplete(long taskID) {
+    setIncomplete(getTaskByID(taskID));
+  }
+
+  public void setIncomplete(TaskData task) {
+    Long parentTaskID = task.getParentTaskID();
+    if (parentTaskID != 0 && getTaskByID(parentTaskID).getStatus() == Status.COMPLETE) {
+      return;
```

Comment:
If there's a failure scenario, try to see if you can have your method return the status.  That way callers can find out if the set method was successful or not.